framework/src/main/java/org/checkerframework/common/wholeprograminference/scenelib/AClassWrapper.java

```diff
@@ -0,0 +1,169 @@
+package org.checkerframework.common.wholeprograminference.scenelib;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import javax.lang.model.element.TypeElement;
+import javax.lang.model.element.VariableElement;
+import javax.lang.model.type.TypeMirror;
+import org.checkerframework.checker.nullness.qual.Nullable;
+import scenelib.annotations.Annotation;
+import scenelib.annotations.el.AClass;
+
+/**
+ * A wrapper for the AClass class from scene-lib that carries additional information that is useful
+ * during WPI.
+ *
+ * <p>This would be better as a subclass of AClass, but it is final.
+ */
+public class AClassWrapper {
+
+    /** The wrapped AClass object. */
+    private AClass theClass;
+
+    /**
+     * The methods of the class. Keys are the names of methods, entries are AMethodWrapper objects.
+     * Wraps the "methods" field of AClass.
+     */
+    private final Map<String, AMethodWrapper> methods = new HashMap<>();
+
+    /**
+     * The fields of the class. Keys are the names of the fields, entries are AFieldWrapper objects.
+     * Wraps the "fields" field of AClass.
+     */
+    private final Map<String, AFieldWrapper> fields = new HashMap<>();
+
+    /** The enum constants of the class, if it has any. */
```

Comment:
It's the enum constants of the class even if the class doesn't have any (it's empty in that case).

Reply:
In a non-enum class, this field is actually null. I've clarified the documentation and added an `@MonotonicNonNull` annotation